src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -164,4 +188,22 @@ private function getTimezone(array $context)
 
         return $dateTimeZone instanceof \DateTimeZone ? $dateTimeZone : new \DateTimeZone($dateTimeZone);
     }
+
+    private function isPreserveContextTimezone(array $context): bool
+    {
+        //Version 5.0 of Symfony will always preserve the context timezone.
+        if (class_exists('\Symfony\Component\HttpKernel\Kernel') && version_compare(Kernel::VERSION, '5.0.00', '>=')) {
+            return true ;
+        }
+
+        if (!isset($context[self::PRESERVE_CONTEXT_TIMEZONE]) && !isset($this->defaultContext[self::PRESERVE_CONTEXT_TIMEZONE])) {
```

Comment:
We should only trigger the deprecation if a timezone was actually provided using `TIMEZONE_KEY` in context or defaultContext.

Reply:
Agreed